rust_programs/ttf_renderer/src/glyphs.rs

```diff
@@ -562,3 +573,107 @@ fn interpret_values_via_flags(
+) -> Vec<isize> {
+    
+    let mut values = vec![];
+
+    if !interpolate {
+        return points[start_index..=end_index].to_vec().clone();
+    }
+    
+    /*
+    for index in start_index..=end_index {
+        values.push(points[index]);
+    }
+    return values;
+    */
+
+    if start_index == end_index {
+        return values;
+    }
+
+    let mut last_value = points[start_index];
+    values.push(last_value);
+    
+    let mut last_on_curve_value = last_value;
+    let mut last_off_curve_value = 0;
+    let mut last_value_was_on_curve = true;
+
+    
+    for index in start_index+1..=end_index {
+        let next_value = points[index];
+        let relative_value = next_value - last_value;
+        last_value = next_value;
+        let flag_set = &flags[index];
+            
+        let next_pt_is_on_curve = flag_set.contains(&GlyphOutlineFlag::OnCurve);
+
+        if last_value_was_on_curve {
+            if next_pt_is_on_curve {
+                // just add new point to the list of points (i.e. draw a
+                // straight line to the new point)
+                last_on_curve_value = next_value;
+                values.push(last_on_curve_value);
+            }
+            else {
+                // the next point is a control point, so we have a 2nd degree
+                // bezier curve, but we don't know the end point yet. Store
+                // the relative value for next round
+                last_off_curve_value = relative_value;
+                last_value_was_on_curve = false;
+            }
+        }
+        else {
+            // Last value was off curve, so we have the (absolute)
+            // start point of the bezier curve in last_on_curve_value and
+            // the (relative) control point in last_off_curve_value.
+            // Need to determine the end point
+            let bezier_start = last_on_curve_value;
+            let bezier_ctrl = last_off_curve_value;
+            let bezier_end = if next_pt_is_on_curve {
+                // next point is on curve, we take it's value as the
+                // (relative) end point
+                relative_value
+            }
+            else {
+                // next point is off curve so we assume an implicit
+                // on-curve point interpolated half way between the
+                // ctrl point and the next off-curve point
+                relative_value / 2
+            };
+            let next_pt = bezier_start + bezier_ctrl + bezier_end;
+
+            
+            values.push(bezier_2d(bezier_start, bezier_ctrl, bezier_end, 0.3));
+            values.push(bezier_2d(bezier_start, bezier_ctrl, bezier_end, 0.6));
+            values.push(next_pt);
+
+            last_on_curve_value = next_pt;
+            last_off_curve_value = if next_pt_is_on_curve {
+                // this next point was on curve, so we're back
+                // to regular straight lines until we see another control point
+                0
```

Comment:
I think this should use `None` instead of using `0` to signify special behaviour

Reply:
Fixed! Definitely need to get over my "c" mindset and remember optionals are available, very cool